Tutorials graphql typescript sdk (#1683)
commercetools/nodejs: docs/sdk/api/graphQLTypescriptSdk.md

```
$ npm install @commercetools/sdk-client @commercetools/sdk-middleware-auth @commercetools/sdk-middleware-http @commercetools/typescript-sdk dotenv
```

Run the program using the following command:
```
$ node project.js
Getting started with commercetools Typescript SDK and GraphQL API
```
If you see the same output as above, we’re ready to start.

### Get commercetools project settings with the commercetools GraphQL API
In this guide, we’ll get project settings information using commercetools GraphQL API. 

Re-open `project.js` and add the following code:
```js
// GraphQL query to get commercetools project settings
const projectSettingsQuery = `
    query {
        project {
        name
        languages
        currencies
        countries
        version
        createdAt
        }
    }
`;

(async () => {
    try {
       await apiRoot.withProjectKey({projectKey}).graphql()
            .post({
                body : {
                    query: projectSettingsQuery,
                    variables: {}
                }
            })
            .execute()
            .then(data => {

>                 console.log('Project information --->', data);

Reviewer: This gives me following output:
`Project information ---> { body: { data: { project: [Object] } }, statusCode: 200 }`
Is there a way to look deeper into the `Object`, like:
```suggestion
                console.log('Project information --->', data.body);
```

Author: @stmeissner We might have to use `JSON.stringify(data)`.

            })
            .catch(error => {
                console.log('ERROR --->', error);
            })
    } catch (error) {
        console.log('ERROR --->', error);
    }
})();
```

> Then the client will **execute** get project information request from `apiRoot` using **TypeScript SDK** and **GraphQL** query[projectSettingsQuery] to get project settings. `projectSettingsQuery` retrieves project's `name`, `languages`,`currencies`,`countries`, `version` and `createdAt` fields. To explore commercetools GraphQL API, you can use an interactive [GraphiQL environment](https://github.com/graphql/graphiql/tree/main/packages/graphiql#readme) available as a part of our [ImpEx & API Playground](https://docs.commercetools.com/docs/login).

Reviewer: ```suggestion
Then the client will **execute** get project information request from `apiRoot` using **TypeScript SDK** and **GraphQL** query[projectSettingsQuery] to get project settings. `projectSettingsQuery` retrieves project's `name`, `languages`,`currencies`,`countries`, `version` and `createdAt` fields. To explore commercetools GraphQL API, you can use an interactive [GraphQL console](https://docs.commercetools.com/api/graphql#interactive-graphql-console).
```

      "lastName": "yourLastName"
    }
  };

// mutation to create new customer
const createCustomerMutation = `
    mutation createCustomer ($newCustomer: CustomerSignUpDraft!) {
        customerSignUp (draft: $newCustomer) {
            customer {
                id
            }
        }
    }
`;

// GraphQL query to get Customer `email` and `firstName`
const getCustomerByIdQuery = `
    query ($id: String) {
        customer (id: $id) {
          email
          firstName
        }
    }
`;
```

`createCustomerMutationVariable` contains the new customer data and mandatory field `email` and `password` and optional fields `firstName` and `lastName`. To find out list of all possible fields you can refer [CustomerDraft](https://docs.commercetools.com/api/projects/customers#customerdraft) documentation. Make sure you pass unique `email` value, you will get an error if try to create a new customer using already existing customer's email. 

`createCustomerMutation` is the GraphQL **mutation** to create a **new customer** on the commercetools project and returns `id`.

`getCustomerByIdQuery` is the GraphQL **query** to get newly created customer info `email` and `firstName` by `id`.

> To explore commercetools GraphQL API you can use an interactive [GraphiQL environment](https://github.com/graphql/graphiql/tree/main/packages/graphiql#readme) which is available as a part of [ImpEx & API Playground](https://docs.commercetools.com/docs/login).

Reviewer: ```suggestion
To explore commercetools GraphQL API you can use an interactive [GraphQL console](https://docs.commercetools.com/api/graphql#interactive-graphql-console).
```

      clientId: CTP_CLIENT_ID,
      clientSecret: CTP_CLIENT_SECRET,
    },
    scopes: [CTP_SCOPES],    fetch,
  })
  const httpMiddleware = createHttpMiddleware({
    host: CTP_API_URL,
    fetch,
  })
  const client = createClient({
    middlewares: [authMiddleware, httpMiddleware],
  })
  // Create an API root from API builder of commercetools platform client
  const apiRoot = createApiBuilderFromCtpClient(client);
  console.log('where clause query with GraphQL and TypeScript SDK');
```

Environment variables and Nodejs dependencies `@commercetools/sdk-client`, `@commercetools/sdk-middleware-auth`,`@commercetools/sdk-middleware-http`, `@commercetools/typescript-sdk`, and `dotenv` are already installed as part of [Getting started with TypeScript SDK and GraphQL](#getting-started) tutorial. Back at the command line, run the program using the following command:
```
$ node whereClauseQuery.js
where clause query with GraphQL and TypeScript SDK
```
If you see the same output as above, we’re ready to start.

### Create GraphQL query and mutation
Add the following code to `whereClauseQuery.js`.

```js
// where clause with query by customer id
const whereClauseCustomerIdVariable = {

>     "where": "id=\"<your-customer-id>\""

Reviewer: ```suggestion
    "where": "email=\"your.test-1@test.com\""
```

Reviewer: this way we could follow up with the "create Customer" example from before.

  };

// GraphQL query to get customer `email` and `firstName` using where clause and customer id query predicate
const getCustomerByWhereClauseQuery = `
    query ($where: String) {
        customers (where: $where) {
          results {
            email
            firstName
          }
        }
    }
`;
```

`whereClauseCustomerIdVariable` contains where clause with value `"id=\"<your-customer-id>\""`. For details about `where` query, refer [Query Predicates](https://docs.commercetools.com/api/predicates/query) documentation. Replace `<your-customer-id>` with your customer [id](https://docs.commercetools.com/api/general-concepts#identifier) for which you would like to retrieve customer `email` and `firstName`.

`getCustomerByWhereClauseQuery` is the GraphQL **query** to get  customer info `email` and `firstName` by using **where clause**  and **customer id** query predicate.

> To explore commercetools GraphQL API, you can use an interactive [GraphiQL environment](https://github.com/graphql/graphiql/tree/main/packages/graphiql#readme) available as a part of [ImpEx & API Playground](https://docs.commercetools.com/docs/login).

Reviewer: ```suggestion
To explore commercetools GraphQL API, you can use an interactive [GraphQL console](https://docs.commercetools.com/api/graphql#interactive-graphql-console).
```
